### PR TITLE
[ready for review] CRM-20952 Refresh membership tab if contribution is membership payment

### PR DESCRIPTION
**Pull request by Edzelopez**
----------------------------------------
* CRM-20952: Refresh membership tab on edits to membership payments
  https://issues.civicrm.org/jira/browse/CRM-20952

### CODE REVIEW

**Comment by Edzelopez** [comment]
@monishdeb or @pradpnayak  can you please QA?

File: CRM/Contribute/Form/AdditionalPayment.php [hunk 1]
```diff
@@ -379,6 +379,12 @@ public function submit($submittedValues) {
     $params = array('id' => $this->_contributionId);
     $contribution = CRM_Contribute_BAO_Contribution::retrieve($params, $defaults, $params);
     CRM_Contribute_BAO_Contribution::addPayments(array($contribution), $contributionStatusId);
+    if ($this->_contributionId && CRM_Core_Permission::access('CiviMember')) {
+      $membershipPayments = civicrm_api3('MembershipPayment', 'Get', array('contribution_id' => $this->_contributionId));
```
Comment (pradpnayak) [inline CRM/Contribute/Form/AdditionalPayment.php]: How about using getcount?

File: CRM/Contribute/Form/Contribution.php [hunk 2]
```diff
@@ -1073,6 +1073,12 @@ public function postProcess() {
     if (empty($this->_id) && !empty($contribution->id)) {
       $this->_id = $contribution->id;
     }
+    if (!empty($this->_id) && CRM_Core_Permission::access('CiviMember')) {
+      $membershipPayments = civicrm_api3('MembershipPayment', 'Get', array('contribution_id' => $this->_id));
```
Comment (pradpnayak) [inline CRM/Contribute/Form/Contribution.php]: same as above.

File: CRM/Contribute/Form/Contribution.php [hunk 3]
```diff
@@ -1073,6 +1073,12 @@ public function postProcess() {
...
+      if ($membershipPayments['count'] > 1) {
```
Comment (pradpnayak) [inline CRM/Contribute/Form/Contribution.php]: this should be > 0 or >=1?

File: CRM/Contribute/Form/AdditionalPayment.php [hunk 4]
```diff
@@ -379,6 +379,12 @@ public function submit($submittedValues) {
...
+      if ($membershipPayments['count'] > 1) {
+        $this->ajaxResponse['updateTabs']['#tab_member'] = CRM_Contact_BAO_Contact::getCountComponent('membership', $this->_contactID);
```
Comment (pradpnayak) [inline CRM/Contribute/Form/AdditionalPayment.php]: same as below.

**Comment by pradpnayak** [comment]
@colemanw , can you please review this?

**Pull request merged**